### PR TITLE
[torch-quant] support different backend on different hardware

### PR DESCRIPTION
**Pull request by chenbohua3**

### CODE REVIEW

**Comment by xiaowan0322** [comment]
Shall we automatically map the calib observers to the corresponding qat observers, by only setting qat observer type (i.e. LSQObserver) ? If so, we can only set act_ob, w_ob, bias_ob and qat_ob  in Quantizer initialization, instead of separately setting in calib and qat functions, which may avoid possible mismatch errors.
For example:
```
ob_ctr = partial(MinMaxObserver, dtype=torch.quint8, qscheme=torch.per_tensor_affine)
DEFAULT_QAT_OB_TYPE = LSQObserver
qat_ob_type = DEFAULT_QAT_OB_TYPE
qat_ob_ctr = partial(
    qat_ob_type,
    dtype=ob_ctr.keywords['dtype'],
    qscheme=ob_ctr.keywords['qscheme'],
)
```

**Comment by chenbohua3** [comment]
> Shall we automatically map the calib observers to the corresponding qat observers, by only setting qat observer type (i.e. LSQObserver) ? If so, we can only set act_ob, w_ob, bias_ob and qat_ob in Quantizer initialization, instead of separately setting in calib and qat functions, which may avoid possible mismatch errors. For example:
> 
> ```
> ob_ctr = partial(MinMaxObserver, dtype=torch.quint8, qscheme=torch.per_tensor_affine)
> DEFAULT_QAT_OB_TYPE = LSQObserver
> qat_ob_type = DEFAULT_QAT_OB_TYPE
> qat_ob_ctr = partial(
>     qat_ob_type,
>     dtype=ob_ctr.keywords['dtype'],
>     qscheme=ob_ctr.keywords['qscheme'],
> )
> ```

done

File: tools/torch_quant/torch_quant/quantizer.py [hunk 1]
```diff
@@ -52,28 +47,129 @@ class Backend(Enum):
...
+    Backend.DISC: partial(DEFAULT_QAT_OB_TYPE, **DEFAULT_AARCH64_ACT_OB_CTR[Backend.DISC].keywords),
+}
+
+
+DEFAULT_GPU_QAT_ACT_OB_CTR: Dict[Backend, Callable[..., Observer]] = {
+    Backend.DISC: partial(DEFAULT_QAT_OB_TYPE, **DEFAULT_GPU_ACT_OB_CTR[Backend.DISC].keywords),
+}
+
+
+DEFAULT_QAT_ACT_OB_CTR = {
```
Comment (xiaowan0322) [inline tools/torch_quant/torch_quant/quantizer.py:130]: Could simplify it a little bit more? Also may avoid some possible mismatch errors.
For example:
```
DEFAULT_QAT_ACT_OB_CTR = {
    device: {
        backend: partial(DEFAULT_QAT_OB_TYPE, **ctr.keywords)
        for backend, ctr in device_ctr.items()
    }
    for device, device_ctr in DEFAULT_ACT_OB_CTR.items()
}
```

File: tools/torch_quant/torch_quant/quantizer.py [hunk 2]
```diff
@@ -52,28 +47,129 @@ class Backend(Enum):
...
+    Backend.DISC: partial(DEFAULT_QAT_OB_TYPE, **DEFAULT_AARCH64_W_OB_CTR[Backend.DISC].keywords),
+}
+
+
+DEFAULT_QAT_GPU_W_OB_CTR: Dict[Backend, Callable[..., Observer]] = {
+    Backend.DISC: partial(DEFAULT_QAT_OB_TYPE, **DEFAULT_GPU_W_OB_CTR[Backend.DISC].keywords),
+}
+
+
+DEFAULT_QAT_W_OB_CTR = {
```
Comment (xiaowan0322) [inline tools/torch_quant/torch_quant/quantizer.py:121]: Same as above

File: tools/torch_quant/torch_quant/quantizer.py (hunk 1 above)
Comment (chenbohua3) [reply tools/torch_quant/torch_quant/quantizer.py:130]: done

**Review by xiaowan0322 — APPROVED** [review]
LGTM

**Pull request merged**